src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -116,11 +107,39 @@ private void PushEvents()
                     _statsd.Gauge(MetricsNames.CpuUserTime, userCpu);
                     _statsd.Gauge(MetricsNames.CpuSystemTime, systemCpu);
                 }
+
+                if (!_exceptionCounts.IsEmpty)
+                {
+                    foreach (var element in _exceptionCounts)
+                    {
+                        _statsd.Increment(MetricsNames.ExceptionsCount, element.Value, tags: new[] { $"exception_type:{element.Key}" });
+                    }
+
+                    _exceptionCounts.Clear();
```

Comment:
I think there's a small (non critical?) race condition here where items can be added to `_exceptionCounts` just before the call to `Clear()`. In that case those new items would be discarded right away and not reported to the backend in the next call to `PushEvents()`.

Reply:
I'm aware of it, yes. I don't know if it's worth the cost to handle it (global locking? allocating a new dictionary?)

Reply:
Actually, allocating a new dictionary will make it less likely but won't completely fix it. I can't think of another way than global locking, and something I would like to avoid as much as possible

Reply:
Can we add a short comment noting that we are aware of it and why we're ok with it?

Reply:
Or we can replace the ConcurrentDictionary by a ConcurrentQueue, but given that we flush only once every 10 seconds I'm afraid of how big it could get. No really, I don't think it's worth the trouble

Reply:
Comment added in https://github.com/DataDog/dd-trace-dotnet/pull/1065/commits/c287e2ce77776b395b21108faa39e038c058be7c